RedHatInsights/rhsm-subscriptions
# SWATCH-2907: Replace custom retry annotation with standard ones

Jira issue: SWATCH-2907

Description
===========

This replaces the workaround we had previously for doing runtime configuration of retry parameters with idiomatic quarkus/smallrye fault tolerance annotations. (This removes the
swatch-common-smallrye-fault-tolerance module).

Our previous pattern was creating high cardinality metrics because the method name wasn't detected properly. With this change, the method name is populated properly, so we can re-enable fault tolerance metrics.

I also removed the env var & template parameter definitions for some of these settings, as we're not using them currently, and we should instead override them in the application.properties file if necessary.

Testing
=======

Setup
-----

1. Temporarily add to `TallySummaryMessageConsumer.consume`:

```java
if (true) {
  throw new RuntimeException("foobar");
}
```

2. Start swatch-billable-usage with the max retries overridden:

```shell
BILLING_PRODUCER_MAX_ATTEMPTS=4 \
./gradlew swatch-billable-usage:quarkusDev
```

Steps
-----

1. Using a kafka producer (IntelliJ kafka producer works well for this), produce a message to `platform.rhsm-subscriptions.tally` with the contents `{"org_id":"foo","tally_snapshots":[]}`.
2. Wait about 15 seconds to allow all retries to finish.
3. Fetch metrics via `http :9000/metrics | grep ft_retry_retries_total`

Verification
------------

1. Notice that the count (the first number in the returned line) is 4.0, corresponding to the retry setting.
2. Notice that the method is specified appropriately.

If you try these steps prior to this change, you should notice that the `method` value is a random UUID.
